### PR TITLE
V0.3.0

### PR DESCRIPTION
**Pull request by sonichi**

### CODE REVIEW

File: flaml/searcher/blendsearch.py [hunk 1]
```diff
@@ -92,15 +93,17 @@ def __init__(self,
...
-        self._points_to_evaluate = points_to_evaluate
+        init_config = low_cost_partial_config or {}
+        self._points_to_evaluate = points_to_evaluate or []
         if global_search_alg is not None:
             self._gs = global_search_alg
         elif getattr(self, '__name__', None) != 'CFO':
             self._gs = GlobalSearch(space=space, metric=metric, mode=mode)
         else:
             self._gs = None
+        # if self._gs is not None and init_config:
```
Comment (qingyun-wu) [inline flaml/searcher/blendsearch.py:104]: remove?

File: flaml/tune/tune.py [hunk 2]
```diff
@@ -148,15 +147,16 @@ def compute_with_config(config):
 
     Args:
         training_function: A user-defined training function. 
-        init_config: A dictionary from a subset of controlled dimensions
-            to the initial low-cost values. e.g.,            
+        config: A dictionary to specify the search space.
+        points_to_evaluate: Initial parameter suggestions to be run first.
```
Comment (qingyun-wu) [inline flaml/tune/tune.py:151]: change to 'points_to_evaluate: A list of initial hyperparameter configuration suggestions to be run first.' ?

File: flaml/searcher/blendsearch.py (hunk 1 above)
Comment (sonichi) [reply flaml/searcher/blendsearch.py]: done

File: flaml/tune/tune.py (hunk 2 above)
Comment (sonichi) [reply flaml/tune/tune.py]: done

**Review by qingyun-wu — APPROVED** [review]

File: flaml/automl.py [hunk 3]
```diff
@@ -642,6 +642,52 @@ def get_estimator_from_log(self, log_file_name, record_id, task):
             )
         return estimator
 
+    def summary_learner_from_log(self, log_file_name, estimator_list):
```
Comment (sonichi) [inline flaml/automl.py:645]: could you respond to the issue and check if this is the desired API?

File: flaml/automl.py [hunk 4]
```diff
@@ -642,6 +642,52 @@ def get_estimator_from_log(self, log_file_name, record_id, task):
...
+        Args:
+            log_file_name: A string of the log file name
+            estimator_list: A list of strings for estimator names, or 'auto'. (same format as that in fit())
+        """
+        class LearnerLogInfo:
+            def __init__(self, learner):
+                self.learner = learner
+                self.iter_num = 0
+                self.val_loss = np.inf
+                self.time_used = 0
```
Comment (sonichi) [inline flaml/automl.py:657]: newline

File: flaml/automl.py [hunk 5]
```diff
@@ -642,6 +642,52 @@ def get_estimator_from_log(self, log_file_name, record_id, task):
             )
         return estimator
 
+    def summary_learner_from_log(self, log_file_name, estimator_list):
+        """Summarize learner info from the specified log file
+
+        Args:
+            log_file_name: A string of the log file name
+            estimator_list: A list of strings for estimator names, or 'auto'. (same format as that in fit())
+        """
```
Comment (sonichi) [inline flaml/automl.py:651]: newline

File: flaml/automl.py [hunk 6]
```diff
@@ -642,6 +642,52 @@ def get_estimator_from_log(self, log_file_name, record_id, task):
...
+                        record.trial_time)
+
+        # print learner info summarized from log.
+        # NOTE about total time in this table: The total time is calculated based
+        # on the recorded trials in the log. Thus It is the actual total time used 
+        # by the learner only when one set log_type='all' in fit()
+        info_name = ['Learner', 'Iteration', 'Val loss', 'Total time']
+        row_format ="{:>15}" * (len(info_name))
+        header = row_format.format(*info_name)
+        print(header)
```
Comment (sonichi) [inline flaml/automl.py:682]: why use both print and logger.info?

Comment (sonichi) [reply flaml/automl.py:682]: again, check in the issue what output is desired.

**Pull request merged**